### PR TITLE
17 custom commands

### LINKED ISSUE #17: Support custom commands
## Acceptance Criteria

1. A project can define its own commands that are registered and made available by the tool

Child of #19

### PR DESCRIPTION
**Pull request by fmizzell**
Fixes #17 
Merge After #25 

**QA**
- Run 'dktl init'
- Create the class given in the README.md into src/command/CustomCommands.php
- Run 'dktl custom:sample'
- Should see "Hello World!!" printed

### CODE REVIEW

File: assets/Command/CustomCommands.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
```
Comment (dafeder) [inline assets/Command/CustomCommands.php]: Let's put this in a documentation example for now

File: assets/composer.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (dafeder) [inline assets/composer.json]: We can still put this in /src it just won't do anything. But let's move it to /src/custom, or something that keeps it close to the command files and doesn't make it seem like it's the _project_ composer.json

File: bin/app.php [hunk 3]
```diff
@@ -1,32 +1,33 @@
...
-$commandClasses = [
-    \DkanTools\Commands\BasicCommands::class,
-    \DkanTools\Commands\DrupalCommands::class,
-    \DkanTools\Commands\DkanCommands::class,
-    \DkanTools\Commands\TestCommands::class,
-    \DkanTools\Commands\DockerCommands::class
-];
+$discovery = new \Consolidation\AnnotatedCommand\CommandFileDiscovery();
+$discovery->setSearchPattern('*Commands.php');
+$commandClasses = $discovery->discover('/usr/local/dkan-tools/src', '\\DkanTools');
```
Comment (dafeder) [inline bin/app.php]: rename vars to
$defaultCommandClasses
$customCommandClasses
$commandClasses

**Comment by dafeder** [comment]
I followed the instructions, and the command doesn't seem to have registered. 

image

**Comment by fmizzell** [comment]
@dafeder The file should be called CustomCommands.php. You are missing the "s" at the end.

**Pull request closed without merging**